### PR TITLE
[#15] Feature/Core Add Database

### LINKED ISSUE #15: Core: Add Database
Previously utilize nedb for the database. We should look to see if anything is newer/better/etc.

### PR DESCRIPTION
**Pull request by LanceGundersen**
## Explanation of work done and why:

This PR adds in core reading and writing "demo" style along with example tests.
Closes #15 

## How to (manually) test:

1. Fetch! `git fetch -p && git checkout -b origin/feature/core-add-database`
2. Serve `yarn electron:serve`
3. Enter a string in the API key input box
image
4. Verify that the input box a) clears the input and b) outputs the api key below
image
5. Reload the window and verify the last API Key is rendered the same as step 4.

## Checklist:
- [x] **Tests written**
image
- [x] **Code linted**
- [x] **Errors handled**
- [x] **Version number bumped**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,21 +27,23 @@
...
-    "@types/mocha": "^5.2.4",
+    "@types/jest": "^23.1.4",
     "@vue/cli-plugin-babel": "^3.11.0",
     "@vue/cli-plugin-typescript": "^3.11.0",
-    "@vue/cli-plugin-unit-mocha": "^3.11.0",
+    "@vue/cli-plugin-unit-jest": "^3.11.0",
     "@vue/cli-service": "^3.11.0",
     "@vue/test-utils": "1.0.0-beta.29",
+    "babel-core": "7.0.0-bridge.0",
     "chai": "^4.1.2",
```
Comment (chrisBosse) [inline package.json]:
```suggestion
```

**Comment by chrisBosse** [comment]
I think `yarn-error.log` should not be included in this PR.

I'm curious how this got past `.gitignore`.

**Comment by LanceGundersen** [comment]
@chrisBosse I noticed the .gitignore was acting really funny. I won't have time till later to take a look but if you wanted to feel free to do a pr to pull in an updated .gitignore. That would help us out.

File: package.json [hunk 2]
```diff
@@ -14,7 +14,9 @@
   },
   "main": "background.js",
   "dependencies": {
+    "@types/lowdb": "^1.0.9",
```
Comment (chrisBosse) [inline package.json]:
```suggestion
```

File: package.json [hunk 3]
```diff
@@ -25,22 +27,23 @@
     "vuex": "^3.0.1"
   },
   "devDependencies": {
-    "@types/chai": "^4.1.0",
-    "@types/mocha": "^5.2.4",
+    "@types/jest": "^23.1.4",
```
Comment (chrisBosse) [inline package.json]:
```suggestion
    "@types/lowdb": "^1.0.9",
    "@types/jest": "^23.1.4",
```

**Comment by chrisBosse** [comment]
@LanceGundersen I recommend running `yarn` (`yarn install`) and committing the updated `yarn.lock` file.

The Yarn lock does not contain Coveralls and still contains Chai.

**Comment by LanceGundersen** [comment]
@chrisBosse gotcha, will do!

**Pull request merged**